src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -92,157 +75,184 @@ internal sealed class ParsedStackFrame : ParsedFrame
                 var members = type.GetMembers();
                 var matchingMembers = members
                     .OfType<IMethodSymbol>()
-                    .Where(m => MemberMatchesMethodName(m, memberName))
+                    .Where(m => m.Name == memberName)
+                    .Where(m => MatchTypeArguments(m.TypeArguments, methodTypeArguments))
+                    .Where(m => MatchParameters(m.Parameters, methodArguments))
                     .ToImmutableArrayOrEmpty();
 
-                if (matchingMembers.Length == 0)
-                {
-                    continue;
-                }
-
                 if (matchingMembers.Length == 1)
                 {
                     return matchingMembers[0];
                 }
             }
 
             return null;
-
-            // TODO: Improve perf here. ToDisplayString is fairly expensive
-            static bool MemberMatchesMethodName(ISymbol member, string memberToSearchFor)
-            {
-                var displayName = member.ToDisplayString();
-                var dotIndex = displayName.LastIndexOf(".");
-                var memberName = dotIndex >= 0
-                    ? displayName[(dotIndex + 1)..]
-                    : displayName;
-
-                return string.Equals(memberName, memberToSearchFor);
-            }
         }
 
         /// <summary>
-        /// Gets all of the text prior to the <see cref="TypeSpan"/>
+        /// If the <see cref="Root"/> has file information, attempts to map it to existing documents
+        /// in a solution. Does fulle filepath match if possible, otherwise does an approximate match
+        /// since the file path may be very different on different machines
         /// </summary>
-        /// <returns></returns>
-        public string GetTextBeforeType()
+        internal (Document? document, int line) GetDocumentAndLine(Solution solution)
         {
-            return OriginalText[..TypeSpan.Start];
-        }
+            var fileMatches = GetFileMatches(solution, out var lineNumber);
+            if (fileMatches.IsEmpty)
+            {
+                return (null, 0);
```

Comment:
nit: i actually prefer `return default;` to indicate you're getting back the 'empty' state.  but this form is also fine if you prefer that.

Reply:
I like it.